### PR TITLE
fix(Open AI): added changes requested by SME

### PR DESCRIPTION
**Pull request by homelessbirds**

### CODE REVIEW

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 1]
```diff
@@ -4,58 +4,65 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
 freshnessValidatedDate: never
 ---
 
-import modelperformancemonitoringOpenAIDashboard from 'images/model-performance-monitoring_screenshot-crop_openai-dashboard.webp'
+import modelperformancemonitoringOpenAIDashboard from 'images/model-performance-monitoring_screenshot-crop_openai-dashboard-1.webp'
 
-With the GPT series application integration, users have the ability to seamlessly monitor OpenAI completion queries and simultaneously log useful statistics in a New Relic customizable dashboard.
+With New Relic's GPT series application integration, you have the ability to seamlessly monitor all OpenAI queries. By adding just two lines of code, your team can gain access to key performance metrics such as cost, response time, rate limits, error rates, chat history, and sample inputs and outputs. Our GPT integration provides real-time metrics to help businesses optimize usage, reduce costs, and achieve better results.
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:9]: I'd only use OpenAI integration rather than GPT app integration, which is hard to parse. Also afaik we don't use New Relic in the possessive for elusive legal reasons

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 2]
```diff
@@ -4,58 +4,65 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
-By adding just two lines of code, users can gain access to key performance metrics such as cost, response time, and sample inputs/outputs. The dashboard also allows users to track total requests, average token/requests, and model names.
-
-With the fully customizable dashboard of New Relic, users can add additional metrics based on their specific business requirements and needs.
-
-Overall, New Relic's GPT integration provides real-time metrics to help businesses optimize usage, reduce costs, and achieve better results.
+This guide will show you how to integrate you OpenAI GPT applications so you can start monitoring your usage.
 
+Once you instrument find OpenAI observability data under **Model Performance** >  **Application**.
 
 <img
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:15]: I'd move this screenshot under your intro paragraph, then as a second paragraph add something like:

> This guide will show you how to integrate you OpenAI GPT applications so you can start monitoring your usage. When you're ready, find your OpenAI data under Model Performance > Application.

Comment (akristen) [reply src/content/docs/mlops/integrations/openai-integration.mdx:15]: So:

Intro paragraph
Screenshot
This snippet of text

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 3]
```diff
@@ -4,58 +4,65 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
+    id="install-with-python" 
+    title="Install with Python"
+  >
 
-## Installation [#installation]
-
-Installation is straightforward and similar to any Python library.
+<Steps>
+<Step>
+### With pip
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:29]: I'd add some kind of intro sentence just so it isn't a free hanging pip instruction. Install blahblahblah using `pip`. 

Here's the GH repo:

https://github.com/newrelic/nr-openai-observability

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 4]
```diff
@@ -4,58 +4,65 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
 <img
   title="OpenAI dashboard"
   alt="OpenAI dashboard"
   src={modelperformancemonitoringOpenAIDashboard}
 />
 
+<CollapserGroup>
+  <Collapser 
+    id="install-with-python" 
+    title="Install with Python"
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:24]: I'm not sure 'with' is the right word here. Maybe 'Install for Python apps'

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 5]
```diff
@@ -4,58 +4,65 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
 `NEW_RELIC_LICENSE_KEY` can also be sent as a parameter at the `monitor.initialization()` call.
 
+If you are reporting data to the New Relic EU region please follow [these](docs/mlops/integrations/openai-integration/#EU-account-users) instructions.
+</Step>
 
-### Step 2. Modify your code [#modify-code]
+<Step>
+### Modify your code [#modify-code]
 
 Add the following two lines to your code:
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:57]: When you reach out to a SME, I would clarify where in the code these snippets should go. I know that sometimes placement matters and it might help customers to know where in their structure these things should appear. (including which files they'll appear in, too)

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 6]
```diff
@@ -4,58 +4,65 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
 from nr_openai_observability import monitor
 monitor.initialization()
 ```
-
-### Step 3. Add your dashboard [#add-dashboard]
+</Step>
+<Step>
+### Add your dashboard [#add-dashboard]
 
 Follow the instruction to add the dashboard to your New Relic account:
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:67]: Same comment as above. What file are they adding this to? A New Relic config file? Their app code?

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 7]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
+    title="Install with Node.js"
+  >
+<Steps>
+<Step>
+### Install OpenAI [#install-openai-node]
+
+Use your favorite package manager and run:
+
+```
+npm install @newrelic/openai-observability openai
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:102]: bash syntax highlighting?

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 8]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
+```
+</Step>
+<Step>
+### Getting started [#getting-started-node]
+
+Call `monitorOpenAI` with the `openAI`instance. 
+
+Here is some example code:
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:110]: same comment as above. where does this appear, etc.

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 9]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
+
+Make sure that you add module type to package.json:
+
+```
+"type": "module"
+```
+</Step>
+
+<Step>
+### Initialization options [#modify-code]
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:143]: It looks like this is the part of the update that has the integration "run" when the app spins up? I think that's how SMEs use initialization. A little more context here would be helpful

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 10]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
+  host?: string;
+  /**
+   * Optional port override for trace endpoint.
+   */
+  port?: number;
+}
+```
+</Step>
+<Step>
+### Environment variables [environment-variables-node]
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:169]:
```suggestion
### Environment variables [#environment-variables-node]
```

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 11]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
+You can use your configured environment variable for initialization options:
+
+`NEW_RELIC_LICENSE_KEY - API`: Key with insert access used to authenticate the request.
+
+`NEW_RELIC_INSERT_KEY`: Same as API key.
+
+`EVENT_CLIENT_HOST`:  Optional host override for events endpoint.
+</Step>
+<Step>
+### Test
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:180]: i'm not really sure what this step does. I think it means 'test that it was set up properly' but i would clarify. i think this is the 'trigger some data and find it in our UI' step

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 12]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
 import os
 os.environ["EVENT_CLIENT_HOST"] = "insights-collector.eu01.nr-data.net"
 ```
 
 `EVENT_CLIENT_HOST` can also be sent as a parameter at the `monitor.initialization()` call.
 
+</Collapser>
 
-## Demonstration video [#demo-walthrough]
+<Collapser
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:213]: I'm not convinced that this video should be in a collapser. I think if you include a video at the end of the doc, letting customers know to jump down to a different section would be helpful

**Review by akristen — COMMENTED** [review]
I think this is a good situation to use our horizontal tabs component, rather than collapsers! i think the collapsers look a little cluttered and the horiz tabs can swap out the relevant install instructions. since Python apps are likely what customers are instrumenting with, consider using that set of instructions as the default :)

File: src/content/docs/mlops/integrations/openai-integration.mdx (hunk 2 above)
Comment (akristen) [reply src/content/docs/mlops/integrations/openai-integration.mdx:15]: idk how this would look though. i'd test it out and choose whatever looks less janky

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 13]
```diff
@@ -75,24 +82,144 @@ openai.Completion.create(
...
+```
+$ npm run test
+```
+It's important to note that  `monitor.integration` test requires `.env.test` file with `OPENAI_API_KEY` configured.
+</Step>
+</Steps>
+
+</Collapser>
+
+<Collapser
```
Comment (akristen) [inline src/content/docs/mlops/integrations/openai-integration.mdx:193]: This feels like it needs to be higher up and not hidden in a collapser.

**Review by RyanKadri — COMMENTED** [review]
A couple suggestions. Also, I generally agree with Alexa's wording suggestions.

Also, to answer where the initialization code should go -- It's pretty flexible. For Node, it just needs to get called before OpenAI calls start coming in. Since it requires the user to pass a configured OpenAI client, it'll have to go after that setup (but that's implied). For python, it needs to come after (not necessarily immediately after) the OpenAI import

File: src/content/docs/mlops/integrations/openai-integration.mdx (hunk 11 above)
Comment (homelessbirds) [reply src/content/docs/mlops/integrations/openai-integration.mdx]: Done

File: src/content/docs/mlops/integrations/openai-integration.mdx (hunk 12 above)
Comment (homelessbirds) [reply src/content/docs/mlops/integrations/openai-integration.mdx]: Done

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 14]
```diff
@@ -75,24 +82,118 @@ openai.Completion.create(
...
+### Install OpenAI [#install-openai-node]
+
+Use npm and run:
+
+```bash
+npm install @newrelic/openai-observability openai
+```
+</Step>
+<Step>
+### Getting started [#getting-started-node]
```
Comment (homelessbirds) [inline src/content/docs/mlops/integrations/openai-integration.mdx:105]: @RyanKadri What else can I name this step? What exactly is happening here?

Comment (RyanKadri) [reply src/content/docs/mlops/integrations/openai-integration.mdx:105]: I see in the python documentation section, you called the corresponding section "Modify your code". I think that sounds fine. Another wording could be something like "Instrument your code". But this section and that one are doing the same thing. Getting the customer to set up our instrumentation library in their codebase

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 15]
```diff
@@ -4,60 +4,67 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
 pip install nr-openai-observability
 ```
+</Step>
+<Step>
+### Set your environment variable [#set-environment-variable]
 
-## Getting Started [#getting-started]
-
-This guide takes you step by step for everything needed to integrate OpenAI GPT applications and start monitoring your usage.
+Get your <InlinePopover type="licenseKey" /> and set it as environment variable: `NEW_RELIC_LICENSE_KEY`.
```
Comment (jeff-colucci) [inline src/content/docs/mlops/integrations/openai-integration.mdx:38]: I think you might need a small lead-in here, it feels kind of abrupt to start the section with "Get your license key". Even just a sentence or a start like "The first thing you need to do" or something along those lines would help soften the abruptness.

File: src/content/docs/mlops/integrations/openai-integration.mdx [hunk 16]
```diff
@@ -4,60 +4,67 @@ metaDescription: "A lightweight tool to monitor your OpenAI workload."
...
+export NEW_RELIC_LICENSE_KEY=<license key>
+```
 
 ```python
 import os
 os.environ["NEW_RELIC_LICENSE_KEY"] = "<license key>"
 ```
 
-`NEW_RELIC_LICENSE_KEY` can also be sent as a parameter at the `monitor.initialization()` call.
+Your `NEW_RELIC_LICENSE_KEY` can also be sent as a parameter at the `monitor.initialization()` call.
```
Comment (jeff-colucci) [inline src/content/docs/mlops/integrations/openai-integration.mdx:51]: 'You can also send your license key as a parameter at'

**Review by jeff-colucci — APPROVED** [review]
I only saw a couple small things but looks great other than that!
